src/vorta/views/partials/treemodel.py

```diff
@@ -994,3 +1042,246 @@ def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
+
+        return parser
+
+    def parse_search_string(self, pattern: str):
+        """
+        Parse the search string into a list of tokens.
+        """
+
+        parser = self.get_parser()
+        return parser.parse_args(pattern.split())
+
+    def filterAcceptsRow(self, sourceRow: int, sourceParent: QModelIndex) -> bool:
+        """
+        Return whether the row should be accepted.
+        """
+
+        if not self.searchPattern:
+            return True
+
+        # Match type "fm" is only available with path
+        if self.searchPattern.match == "fm" and not self.searchPattern.path:
+            self.searchStringError.emit(True)
+            return False
+
+        model = self.sourceModel()
+        item = model.index(sourceRow, 0, sourceParent).internalPointer()
+
+        item_path = path_to_str(item.path)
+        item_name = item.subpath
+
+        # Exclude Parents
+        if self.searchPattern.exclude_parents:
+            if item.children:
+                return False
+
+        # Set default values
+        if self.searchPattern.match is None:
+            self.searchPattern.match = "fm" if self.searchPattern.path else "in"
+
+        if self.searchPattern.path:
+            search_item = item_path
+        else:
+            search_item = item_name
+
+        if self.searchPattern.search_string:
+            # TODO: Move operations on search string to setSearchString method
+
+            search_string = " ".join(self.searchPattern.search_string)
+
+            # Ignore Case?
+            if self.searchPattern.ignore_case:
+                search_item = search_item.lower()
+                search_string = search_string.lower()
+
+            if self.searchPattern.match == "in" and search_string not in search_item:
+                return False
+            elif self.searchPattern.match == "ex" and search_string != search_item:
+                return False
+            elif self.searchPattern.match == "re" and not re.search(search_string, search_item):
+                return False
+            elif self.searchPattern.match == "fm" and not fnmatch(search_item, search_string):
+                return False
+
+        if self.searchPattern.size:
+            # Diff view has size column corresponding to the changed_size while
+            # Extract view has size column corresponding to the size
+            if hasattr(item.data, 'changed_size'):
+                item_size = item.data.changed_size
+            else:
+                item_size = item.data.size
+
+            for filter_size in self.searchPattern.size:
+                if not compare_values_with_sign(item_size, filter_size[1], filter_size[0]):
+                    return False
+
+        if self.searchPattern.change:
+            item_change = item.data.change_type.short()
+
+            if item_change != self.searchPattern.change:
+                return False
+
+        return True
+
+
+class DiffTreeSortFilterProxyModel(FileTreeSortFilterProxyModel):
```

Comment:
I think you don't need to have `DiffTreeSortProxyModel` and `DiffTreeSortFilterProxyModel`. You can merge the methods into one class. Or am I overlooking an advantage?

Reply:
@real-yfprojects I couldn't find where `DiffTreeSortProxyModel` is defined. I only see `DiffTreeSortFilterProxyModel`

Reply:
In diff_result.py

Reply:
@real-yfprojects I still can't find any mention. Can you check the PR changes once again?

Reply:
https://github.com/borgbase/vorta/blob/7ebc36ee733934a38f7ab6405f3287d12f3908ee/src/vorta/views/diff_result.py#L396

Reply:
I'm getting circular import issue for `ChangeType` class import in `treemodel.py` from `diff_result.py`. Should I move it inside treemodel and import in the latter one?

Reply:
If you only keep https://github.com/borgbase/vorta/blob/7ebc36ee733934a38f7ab6405f3287d12f3908ee/src/vorta/views/diff_result.py#L396
and add the code from https://github.com/borgbase/vorta/blob/8950eb2c94f8abb1a3b14b1413ebd78e40263b01/src/vorta/views/partials/treemodel.py#L1160 to it. You shouldn't get any circular imports.

Reply:
Please check the implementation in next commit.